assembly/sdk/storage.ts

```diff
@@ -241,6 +241,79 @@ export class Storage {
     }
   }
 
+  /**
+   * Reads value of key into register.
+   *
+   * ```ts
+   * storage.read_raw("myKey", 1);
+   * ```
+   *
+   * @param key The unique identifier associated with a value in a key-value store
+   * @param register_id The ID of the register to store the value of key into
+   * @returns An errorcode. A value of `1` means success.
+   */
+  static read_raw(key: string, register_id: u64): u64 {
+    let key_encoded = util.stringToBytes(key);
+    return env.storage_read(
+      key_encoded.byteLength,
+      key_encoded.dataStart,
+      register_id
+    );
+  }
+
+  /**
+   * Writes value of given register into the storage under the key.
+   *
+   * If an old value was present under key, it is swapped with the new value in register.
+   *
+   * ```ts
+   * storage.write_key("myKey", 1);
+   * ```
+   *
+   * @param key The unique identifier associated with a value in a key-value store
+   * @param register_id The ID of the register from which the value is obtained
+   * @returns An errorcode. A value of `1` means success.
+   */
+  static write_raw(key: string, register_id: u64): u64 {
+    let key_encoded = util.stringToBytes(key);
+    let res = env.storage_write(
+      key_encoded.byteLength,
+      key_encoded.dataStart,
+      0,
+      register_id,
+      register_id
+    );
+    env.value_return(0, register_id);
```

Comment:
Why are you returning?

From the instructions:
```ts

namespace env {
  ...
  export function return_raw(register_id: u64): void; //Hint in above example.
}
```
The example was how to write `return_raw` only.  This function and the other only write to storage they do not return anything.

Reply:
Is `return_raw` actually used for anything then? Because I can't seem to see where it can be used.

Reply:
You're just supposed add it as a function in the namespace.  Currently when you return a value the bindings generated calls to `value_return`, which then passes the memory that holds the serialized data.  You are adding this function so that users have a convenient way to return the contents of a register instead of the memory.  This saves on gas as you never have to copy the data to and from memory.